### DISCUSSION TITLE
correct way to move OneDrive folder

### DISCUSSION THREAD

**Discussion by ccouto** [abraunegg/onedrive#2683] [General]
What is the correct way to move OneDrive folder? For example, my OneDrive is currently on /home/ccouto/OneDrive and I would like to move it to /home/ccouto/Documents/Onedrive.

Should I just move the folder using the OS "mv" command and change the config file to point to the new directory?

**Comment by abraunegg** [comment]
@ccouto 
> What is the correct way to move OneDrive folder?

The steps you have outlined are the most correct way to do this, however I would add to stop any systemd 'onedrive' background service that may be running, validate that there is no active 'onedrive' process before taking such action.

Once you modify your 'config' file, make sure you validate your configuration with `--display-config` to ensure that the application is going to use your new location correctly.

Performing this action will then also trigger a `--resync` requirement as the 'sync_dir' has changed when you do restart any application use.

**Reply by mc2contributor** [reply]
Why must we resync when the sync_dir location changes?  The contents haven't changed, so we should still be in sync.  Would it be easy to provide an option to override the need for resync in a case like this?

**Reply by mc2contributor** [reply]
I figured out that I just need to copy the config file to .config.backup to stop onedrive from demanding a resync for a moved folder.  I'm glad it's that easy :-)

**Comment by abraunegg** [comment]
@mc2contributor 
> Why must we resync when the sync_dir location changes? The contents haven't changed, so we should still be in sync.

The reason is simple and historic issues that people have had

Folk when changing the sync_dir option sometimes do this by just making a new (empty) directory and then reconfiguring the application.

Without forcing a `--resync` the client would see this empty directory and *assume* you deleted everything locally, thus, delete things online ...

This is why this is needed - to protect users from shooting themselves in their feet.
